_usage/query_planning.md

```diff
@@ -0,0 +1,156 @@
+---
+layout: default
+title: Reviewing a Query Plan
+nav_order: 8
+---
+
+## Reviewing a Query Plan
+
+Babelfish version 2.1.0 includes support for functions and parameters that allow you to review estimated and actual (executed) query plans. This allows you to use the TDS port to identify and refine slow-performing queries. The functions can display an query plan for:
+
+- `SELECT`, `INSERT`, `UPDATE`, and `DELETE` statements
+- Nested procedures, multiple statements in a batch
+- Variable declarations (for example, `DECLARE @t TABLE (a int, b int);`)
+
+The functions *do not* support query plan retrieval from functions, control flows, and cursors.
+
+You can use a SET statement on the TDS port to turn on/off the following functions: 
+
+- SET BABELFISH_STATISTICS PROFILE {ON|OFF} to display the query plan used to execute a statement. The command implements the behavior of the [PostgreSQL EXPLAIN ANALYZE statement] (https://www.postgresql.org/docs/current/using-explain.html#USING-EXPLAIN-ANALYZE).
+    
+- SET BABELFISH_SHOWPLAN_ALL {ON|OFF} to display estimated execution plans for a statement without performing the command. The command implements the behavior of the [PostgreSQL EXPLAIN statement](https://www.postgresql.org/docs/current/using-explain.html#USING-EXPLAIN-BASICS).
+
+For example, the following command sequence turns on query planning, returns the query plan for the SELECT statement without executing the command, and then turns off query planning:
+
+```
+SET BABELFISH_SHOWPLAN_ALL ON
+GO
+SELECT b.bid, b.bbalance, t.tid, t.tbalance FROM pgbench_branches b, pgbench_tellers t WHERE b.bid=t.bid ORDER BY b.bid;
+GO
+                                     QUERY PLAN                                      
+-------------------------------------------------------------------------------------
+ Sort  (cost=37.96..39.21 rows=500 width=16)
+   Sort Key: b.bid
+   ->  Hash Join  (cost=3.12..15.55 rows=500 width=16)
+         Hash Cond: (t.bid = b.bid)
+         ->  Seq Scan on pgbench_tellers t  (cost=0.00..11.00 rows=500 width=12)
+         ->  Hash  (cost=2.50..2.50 rows=50 width=8)
+               ->  Seq Scan on pgbench_branches b  (cost=0.00..2.50 rows=50 width=8)
+(7 rows)
+
+SET BABELFISH_SHOWPLAN_ALL OFF
+GO
+```
+
+If you use the BABELFISH_STATISTICS PROFILE function, Babelfish will display the query plan that provides the best performance as it invokes the SELECT statement.  For example, the following command sequence returns an estimated cost to execute the SELECT statement and then turns off query planning:
```

Comment:
this is not a function

Reply:
I've also noted that it's not the query plan that provides the best performance.

Reply:
I've also noted that it's not the query plan that provides the best performance.